tensorrt_llm/_torch/attention_backend/sparse/dsa.py

```diff
@@ -520,8 +542,98 @@ def update_spec_dec_param(
             capture_graph = torch.cuda.is_current_stream_capturing()
             self.create_expanded_buffers(capture_graph=capture_graph)
 
+    def prepare_dense_topk_indices(self,
+                                   kv_lens,
+                                   device=False):  # device=False means use CPU
+
+        @maybe_compile(dynamic=True)
+        def _get_dense_topk_indices(seq_lens, kv_lens, num_tokens):
```

Comment:
This may be a bit irrelevant and you can consider as non-blocking comment for this PR:
Why do we need this func `_get_dense_topk_indices` at all if we decided to skip indexer module (except indexer.update_k)? Why not just use our existing (full-attn)MLA kernel (so that we can avoid extra gather op in the attn op)?